File: clients/clients-service/src/main/java/com/hyperbench/clients/ClientsApplication.java
package com.hyperbench.clients;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.ComponentScan;

@SpringBootApplication
@ComponentScan(basePackages = {"com.hyperbench.clients", "com.hyperbench.shared"})
public class ClientsApplication {

    public static void main(String[] args) {
        SpringApplication.run(ClientsApplication.class, args);
    }
}

File: clients/clients-service/src/main/kotlin/com/hyperbench/clients/ClientsApplication.kt
package com.hyperbench.clients

import com.hyperbench.halschemaforms.mediatype.HalSchemaFormsConfiguration
import com.hyperbench.shared.context.WorkbenchContextFilter
import com.hyperbench.clients.config.HalFormsConfig
import com.hyperbench.clients.config.WebConfig
import com.hyperbench.clients.controller.ClientUiController
import org.springframework.boot.autoconfigure.EnableAutoConfiguration
import org.springframework.boot.SpringBootConfiguration
import org.springframework.boot.runApplication
import org.springframework.context.support.beans

@SpringBootConfiguration(proxyBeanMethods = false)
@EnableAutoConfiguration
class ClientsApplication

val beans = beans {
    bean<WorkbenchContextFilter>()
    bean<HalSchemaFormsConfiguration>()
    bean<WebConfig>()
    bean<HalFormsConfig>()
    bean<ClientUiController>()
}

fun main(args: Array<String>) {
    runApplication<ClientsApplication>(*args) {
        addInitializers(beans)
    }
}
